plugins/relay/src/server.c

```diff
@@ -209,7 +209,7 @@ eventd_relay_server_new(void)
     server = g_new0(EventdRelayServer, 1);
 
     server->evp = libeventd_evp_context_new(server, &_eventd_relay_interface);
-    server->reconnect = libeventd_reconnect_new(5, 10,_eventd_relay_reconnect_callback, server);
+    server->reconnect = libeventd_reconnect_new(5, 10,_eventd_relay_reconnect_callback, server, g_free);
```

Comment:
`server` is always inserted in the plugin context `GHashTable *servers`, which is created using `g_hash_table_new_full(g_str_hash, g_str_equal, g_free, eventd_relay_server_free);` (line 73) so you have 2 mistakes here:

1. The data will be double-freed
2. You are using the wrong free function